Update time management for move to chrony (#1380)
balena-io/docs: pages/reference/OS/time.md

---
# Time management

__Note:__ Starting from {{ $names.os.lower }} 2.13.1 the `chrony` service is used for time management. Prior versions of {{ $names.os.lower }} use `systemd-timesyncd`.

Devices running {{ $names.os.lower }} make use of the `chrony` (or `systemd-timesyncd`) service to keep the system time synchronized. That service is running in the host OS, independent of the application containers.

It is important that the date and time are set correctly, as an inaccurate date can manifest itself as several different issues on the device, such as SSL/TLS certificates appearing invalid. If you want to query the current time on the device, you can do so by using the `date` utility or the datetime related functions of the standard library of your language.

```shell
$ date
Tue Mar  3 19:59:54 UTC 2020
```

> {{ $names.company.upper }} devices use [UTC](https://en.wikipedia.org/wiki/Coordinated_Universal_Time) as their standard time zone. If you want to use a timezone other than UTC, you have to explicitly set the timezone using the tools available for the operating system running within your container. The [balena-timezone]({{ $links.githubPlayground }}/balena-timezone) project is an example of how to do this in Debian. The {{ $names.cloud.lower }} dashboard shows log timestamps using the local time for the computer you are viewing it on, and you can switch it to display the timestamps in UTC by using the toggle available in the log view.

Reviewer: ```suggestion
{{ $names.company.upper }} devices use [UTC](https://en.wikipedia.org/wiki/Coordinated_Universal_Time) as their standard time zone. If you want to use a timezone other than UTC, you have to explicitly set the timezone using the tools available for the operating system running within your container. The [balena-timezone]({{ $links.githubPlayground }}/balena-timezone) project is an example of how to do this in Debian. The {{ $names.cloud.lower }} dashboard shows log timestamps using the local time for the browser you are viewing it on, and you can switch it to display the timestamps in UTC by using the toggle available in the log view.
```

![Change the logs to use UTC timezone](/img/common/main_dashboard/dashboard-utc.png)

If you want to learn if the system has completed at least one successful NTP synchronization since boot, you can use D-Bus from within your container. Check the "[Checking if device time is NTP synchronized](/runtime/runtime/#checking-if-device-time-is-ntp-synchronized)" example. The `NTPSynchronized` property will indicate if the device is running with a potentially stale clock or if the system time is synced.

## chrony

> [chrony][chrony] is an implementation of the [Network Time Protocol (NTP)](https://en.wikipedia.org/wiki/Network_Time_Protocol). The chrony daemon `chronyd` is [configured](https://chrony.tuxfamily.org/doc/3.5/chrony.conf.html) via the `/etc/chrony.conf` file.  The default `chrony.conf` file used in {{$names.os.lower}} can be viewed [here](https://github.com/balena-os/meta-balena/blob/master/meta-balena-common/recipes-core/chrony/files/chrony.conf).

Reviewer: ```suggestion
[chrony][chrony] is an implementation of the [Network Time Protocol (NTP)](https://en.wikipedia.org/wiki/Network_Time_Protocol). The chrony daemon `chronyd` is [configured](https://chrony.tuxfamily.org/doc/3.5/chrony.conf.html) via the `/etc/chrony.conf` file.  The default `chrony.conf` file used in {{$names.os.lower}} can be viewed [here]({{ $links.githubOS }}/meta-balena/blob/master/meta-balena-common/recipes-core/chrony/files/chrony.conf).
```

When you first provision a device, as a fallback, `/var/lib/systemd/clock` is set to the timestamp of the host OS build (or more precisely, the timestamp of the systemd build within the host OS). For more info, you can check the [`systemd-timesyncd` documentation](https://www.freedesktop.org/software/systemd/man/systemd-timesyncd.service.html) and the [timesyncd source code](https://github.com/systemd/systemd/blob/master/src/timesync/timesyncd.c).

## Networking Requirements

There are certain networking requirements to ensure that the NTP service can properly function, and the device time may be kept synchronized.

> The NTP service requires UDP port `123` to be open. See more at our [network requirements page](/deployment/network/2.0.0/#network-requirements).

Reviewer: ```suggestion
The NTP service requires UDP port `123` to be open. See more [network requirements here](/deployment/network/2.0.0/#network-requirements).
```
